full-service/src/db/transaction_log.rs

```diff
@@ -802,13 +803,32 @@ impl TransactionLogModel for TransactionLog {
         Ok(())
     }
 
+    fn update_consumed_txo_to_failed(
+        transaction_input_txo_id_hex: &str,
+        conn: Conn,
+    ) -> Result<(), WalletDbError> {
+        let transaction_log_ids: Vec<String> = transaction_logs::table
+            .inner_join(transaction_input_txos::table)
+            .filter(transaction_input_txos::txo_id.eq(transaction_input_txo_id_hex))
```

Comment:
should there be a filter here to make sure that the transaction_log was submitted so that logs for unsubmitted transactions don't become confused with ones that were submitted and failed?

Or is this only reached for submitted transaction_logs already?

Reply:
since this is coming in via input txos, not transaction logs, this will grab any and all transaction logs that have the input txo
¿Do we have a use case where one would build a transaction log and want to see it in the built state even though it can never succeed?

In particular when we saw the spent key image error, it was because 
1. the transaction log was built.
2. A different transaction log with the same input TXOs landed on the block chain
3. the first transaction log was submitted, because it was showing up as `built` not `failed`
4. The consensus nodes kept rejecting the transaction log due to key images already spent

Reply:
The use case that I'm thinking through is when a client, e.g. Desktop Wallet, does a build_transaction prior to asking the user to confirm the transaction, and then the user declines to confirm.

This will leave a transaction_log with a submitted_block_index of null

I *think* for error reporting in the client, it would be good for the client to be able to tell the difference between unsubmitted transactions and failed transactions.

Reply:
do we already set these to failed when the tombstone block passes?

Reply:
it looks like those transaction logs will be marked failed once the tombstone block is exceeded, [update_pending_exceeding_tombstone_block_index_to_failed](https://github.com/mobilecoinofficial/full-service/blob/cb2f46e3d6c03e9e06edc15bd98fa08bf9c0b90c/full-service/src/db/transaction_log.rs#L805)
```rust
        diesel::update(
            transaction_logs::table
                .filter(transaction_logs::account_id.eq(&account_id.0))
                .filter(transaction_logs::tombstone_block_index.lt(block_index as i64))
                .filter(transaction_logs::failed.eq(false))
                .filter(transaction_logs::finalized_block_index.is_null()),
        )
        .set((transaction_logs::failed.eq(true),))
        .execute(conn)?;
```

Reply:
Which jives with the experience we had where we kept trying to submit the transaction, and the consensus node kept saying already spent
our service saw the transaction log in the build state, until the tombstone block was exceeded